### ISSUE TITLE
btrfs filesystem du does not correctly report disk usage

### ISSUE THREAD

**Issue by crass** [kdave/btrfs-progs#518]
I would expect `btrfs fi du` to show a somewhat accurate picture of how much of the disk is actually used by a file or directory tree. However, I don't think this is what du measures.

Take this output:
```
# btrfs fi du disk.img
     Total   Exclusive  Set shared  Filename
 146.02GiB     8.00KiB   146.02GiB  disk.img
# compsize disk.img
Processed 1 file, 782439 regular extents (888382 refs), 0 inline.
Type       Perc     Disk Usage   Uncompressed Referenced  
TOTAL       62%       83G         134G         146G       
none       100%       57G          57G          58G       
zstd        34%       26G          77G          87G
```

As can be seen here `compsize` shows that the file is using only 83G while `btrfs fi du` shows a total usage of 146G. It appears as though du's output is equivalent to the `Referenced` column of `compsize` output. What  does du's total actually mean? (eg. size of all extents when not compressed?)

Ultimately, I think it would be nice to have options for du to display compressed/uncompressed output for  similar to compsize.

**Comment by kdave** [comment]
Agreed, the compression similar to what compsize does could be added to the `fi du` output. The purpose of 'du' is to show "data usage", on the logical level so the referenced/shared/compresssed are the numbers that should appear. I thought if the physical extent placement on devices should be also printed within 'du' but that's may be a overloading one command and another would be more suitable.
